src/main/java/com/laserfiche/api/client/APIServer/TokenClientImpl.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.laserfiche.api.client.model.ApiException;
+import com.laserfiche.api.client.model.CreateConnectionRequest;
+import com.laserfiche.api.client.model.OffsetDateTimeDeserializer;
+import com.laserfiche.api.client.model.ProblemDetails;
+import com.laserfiche.api.client.oauth.TokenClientObjectMapper;
+import kong.unirest.Header;
+import kong.unirest.HttpResponse;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONObject;
+import org.threeten.bp.OffsetDateTime;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class TokenClientImpl implements TokenClient {
+    private String baseUrl;
+
+    protected ObjectMapper objectMapper;
+
+    public TokenClientImpl(String baseUrl) {
+        this.baseUrl = baseUrl;
+        Unirest
+                .config()
+                .setObjectMapper(new TokenClientObjectMapper());
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(OffsetDateTime.class, new OffsetDateTimeDeserializer());
+        this.objectMapper = JsonMapper
+                .builder()
+                .addModule(module)
+                .disable(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS)
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+                .build();
+    }
+
+    @Override
+    public CompletableFuture<SessionKeyInfo> createAccessToken(String repoId, CreateConnectionRequest body) {
+        Map<String, Object> pathParameters = getNonNullParameters(new String[]{"repoId"},
+                new Object[]{repoId});
+        CompletableFuture<HttpResponse<SessionKeyInfo>> future = Unirest
+                .post(baseUrl + "/v1/Repositories/{repoId}/Token")
+                .routeParam(pathParameters)
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .field("grant_type", "password")
+                .field("username", body
+                        .getUsername()
+                        .toString())
+                .field("password", body
+                        .getPassword()
+                        .toString())
+                .asObjectAsync(SessionKeyInfo.class);
+        return future.thenApply(httpResponse -> {
+            if (httpResponse.getStatus() != 200) {
+                ProblemDetails problemDetails;
+                try {
+                    String jsonString = new JSONObject(httpResponse.getBody()).toString();
+                    problemDetails = objectMapper.readValue(jsonString, ProblemDetails.class);
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
+                    return null;
```

Comment:
Not sure if we should terminate execution here. Like, maybe we should just throw the exception and not to return a null?

Reply:
Maybe it's okay as is.

Reply:
updated to new version yesterday